File: leetcode/minimum_time_visiting_all_points.py
#
# Link: https://leetcode.com/problems/minimum-time-visiting-all-points
#


#
# Solution 1
#
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        result = 0
        pointsLen = len(points)

        for i in range(0, pointsLen - 1):
            [x1, y1] = points[i]
            [x2, y2] = points[i + 1]

            result += max(abs(x2 - x1), abs(y2 - y1))

        return result

File: leetcode/minimum_time_visiting_all_points.go
//
// Link: https://leetcode.com/problems/minimum-time-visiting-all-points
//

//
// Solution 1
//
func minTimeToVisitAllPoints(points [][]int) int {
	result := 0;
	pointsLen := len(points);

	for i := 0; i < pointsLen-1; i++ {
		x1 := points[i][0];
		y1 := points[i][1];

		x2 := points[i+1][0];
		y2 := points[i+1][1];

		result += int(math.Max(
			float64(math.Abs(float64(x2-x1))),
			float64(math.Abs(float64(y2-y1))),
		));
	};

	return result;
};
